File: 2024-05-27(server)/project-2/src/main/java/com/example/app/service/UserService.java
package com.example.app.service;

import com.example.app.model.User;
import com.example.app.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class UserService {

    @Autowired
    private UserRepository userRepository;

    public List<User> getAllUsers() {
        return userRepository.findAll();
    }

    public User getUserById(Long id) {
        return userRepository.findById(id)
                             .orElseThrow(() -> new RuntimeException("User not found with id: " + id));
    }

    public User saveUser(User user) {
        return userRepository.save(user);
    }

    public User updateUser(Long id, User updatedUser) {
        User existingUser = getUserById(id);
        existingUser.setUsername(updatedUser.getUsername());
        existingUser.setPassword(updatedUser.getPassword());
        existingUser.setLat(updatedUser.getLat());
        existingUser.setLng(updatedUser.getLng());
        existingUser.setPhone(updatedUser.getPhone());
        existingUser.setRole(updatedUser.getRole());
        return userRepository.save(existingUser);
    }
    public User getUserByUsername(String username) {
        return userRepository.findByUsername(username);
    }
    public void deleteUser(Long id) {
        userRepository.deleteById(id);
    }

}

File: 2024-05-28/MyActivity2/app/src/main/java/com/example/myactivity/data/service/UserService.kt
package com.example.myactivity.data.service

import com.example.myactivity.data.model.User
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.DELETE
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Path

interface UserService {
    // user endpoints
    @GET("/api/users")
    fun getAllUsers(): Call<List<User>>

    @GET("/api/users/{id}")
    fun getUserById(@Path("id") id: Int): Call<User>

    @POST("/api/users")
    fun createUser(@Body user: User): Call<User>

    @PUT("/api/users/{id}")
    fun updateUser(@Path("id") id: Int, @Body user: User): Call<User>

    @DELETE("/api/users/{id}")
    fun deleteUser(@Path("id") id: Int): Call<Void>
}